azurerm/resource_arm_managed_cluster.go

```diff
@@ -0,0 +1,556 @@
+						},
+
+						"vnet_subnet_id": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+
+						"os_type": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  containerservice.Linux,
+						},
+					},
+				},
+				Set: resourceAzureRMManagedClusterAgentPoolProfilesHash,
+			},
+
+			"service_principal": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"client_id": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"client_secret": {
+							Type:      schema.TypeString,
+							Required:  true,
+							Sensitive: true,
+						},
+					},
+				},
+				Set: resourceAzureRMManagedClusterServicePrincipalProfileHash,
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceArmManagedClusterCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient)
+	managedClusterClient := client.managedClustersClient
+
+	log.Printf("[INFO] preparing arguments for Azure ARM AKS managed cluster creation.")
+
+	resGroup := d.Get("resource_group_name").(string)
+	name := d.Get("name").(string)
+	location := d.Get("location").(string)
+	dnsPrefix := d.Get("dns_prefix").(string)
+	kubernetesVersion := d.Get("kubernetes_version").(string)
+
+	linuxProfile := expandAzureRmManagedClusterLinuxProfile(d)
+	agentProfiles := expandAzureRmManagedClusterAgentProfiles(d)
+
+	tags := d.Get("tags").(map[string]interface{})
+
+	parameters := containerservice.ManagedCluster{
+		Name:     &name,
+		Location: &location,
+		ManagedClusterProperties: &containerservice.ManagedClusterProperties{
+			DNSPrefix:         &dnsPrefix,
+			KubernetesVersion: &kubernetesVersion,
+			LinuxProfile:      &linuxProfile,
+			AgentPoolProfiles: &agentProfiles,
+		},
+		Tags: expandTags(tags),
+	}
+
+	servicePrincipalProfile := expandAzureRmManagedClusterServicePrincipal(d)
+	if servicePrincipalProfile != nil {
+		parameters.ServicePrincipalProfile = servicePrincipalProfile
+	}
+
+	ctx := client.StopContext
+	_, error := managedClusterClient.CreateOrUpdate(ctx, resGroup, name, parameters)
+	if error != nil {
+		return error
+	}
+
+	read, err := managedClusterClient.Get(ctx, resGroup, name)
+	if err != nil {
+		return err
+	}
+
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read AKS managed cluster %s (resource group %s) ID", name, resGroup)
+	}
+
+	log.Printf("[DEBUG] Waiting for AKS managed cluster (%s) to become available", d.Get("name"))
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"Updating", "Creating"},
+		Target:     []string{"Succeeded"},
+		Refresh:    ManagedClusterStateRefreshFunc(client, resGroup, name),
+		Timeout:    30 * time.Minute,
+		MinTimeout: 15 * time.Second,
+	}
+	if _, err := stateConf.WaitForState(); err != nil {
+		return fmt.Errorf("Error waiting for AKS managed cluster (%s) to become available: %s", d.Get("name"), err)
+	}
```

Comment:
so we should be able to remove this in favour of using the Future returned from the `CreateOrUpdate` above

Reply:
Thanks, fixed.